servicetalk-tcp-netty-internal/src/main/java/io/servicetalk/tcp/netty/internal/TcpServerConfig.java

```diff
@@ -95,9 +126,10 @@ public void transportObserver(final TransportObserver transportObserver) {
      * found the corresponding {@link ServerSslConfig} is used.
      * @return {@code this}
      */
-    public TcpServerConfig sslConfig(ServerSslConfig defaultSslConfig, Map<String, ServerSslConfig> sniConfig) {
+    public TcpServerConfig sslConfig(@Nullable ServerSslConfig defaultSslConfig,
+                                     @Nullable Map<String, ServerSslConfig> sniConfig) {
         sslConfig(defaultSslConfig);
-        this.sniConfig = requireNonNull(sniConfig);
+        this.sniConfig = sniConfig;
         return this;
```

Comment:
What happens if we call `.sslConfig(config1, true).sslConfig(config2)`? I would expect to find a final state of `.sslConfig() == config2 && .acceptInsecureConnections() == false`, but I think we'll find `.sslConfig() == config2 && .acceptInsecureConnections() == true` right now. Since `DefaultHttpServerBuilder` defers to `TcpServerConfig`, I think it inherits this behavior and is probably a security foot-gun.

Reply:
Good point! I would expect the same.
I think it should be backward compatible if we remove `sslConfig` method from `AbstractTcpConfig` (because it's a pkg-private abstract class) and implement it on `TcpServerConfig/TcpClientConfig`, where `TcpServerConfig` will reset the flag to `false`.

@daschl this is how you can validate that with japicmp in your branch:
```
./gradlew assemble
./scripts/japicmp.sh 0.42.42 local
```